### PR TITLE
Add API for commits

### PR DESCRIPTION
**Pull request by christophemaximin**
Gumroad's deployment process is a bit cumbersome, so I'm considering making little script to, in one command:
- select all the verified commits on staging up to the last unverified commit (if any)
- tell you what they are
- run the appropriate commands to deploy them in production and update the production branch

Eventually it could be updated to make hotfixes easier and clearer as to what they include and what they skip. 

This PR is necessary for that script to know which commits are verified and which aren't.

### CODE REVIEW

File: app/controllers/api/commits_controller.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+class Api::CommitsController < ApiController
+  before_action :select_repo
+
+  def index
+    commits = @repo.commits.limit(50).reorder(id: VALID_SORT_DIRECTIONS[params[:sort_direction]])
+    commits = commits.unverified if params[:filter] == "unverified"
+    commits = commits.verified if params[:filter] == "verified"
+
+    serialized_commits = commits.map { |commit| serialize(commit) }
+    render json: { commits: serialized_commits }
```
Comment (slavingia) [inline app/controllers/api/commits_controller.rb]: maybe a status: 'success' here?

File: app/controllers/api/commits_controller.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  end
+
+  private
+
+  def serialize(commit)
```
Comment (slavingia) [inline app/controllers/api/commits_controller.rb]: commit.as_json method?

File: app/controllers/api/commits_controller.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    {
+      id: commit.id,
+      sha1: commit.sha1,
+      author_name: commit.author_name,
+      message: commit.message,
+    }
+  end
+
+  def select_repo
```
Comment (slavingia) [inline app/controllers/api/commits_controller.rb]: fetch_repo? (i think rubocop prefers that)

File: app/controllers/api/commits_controller.rb (hunk 2 above)
Comment (christophemaximin) [reply app/controllers/api/commits_controller.rb]: I didn't necessarily want to return all columns but sure, I rendered the AR collection directly as the json conversion is implicit anyway

**Pull request merged**